src/test/java/com/mysql/cj/jdbc/ha/ca/plugins/ConnectionPluginManagerTest.java

```diff
@@ -0,0 +1,82 @@
+package com.mysql.cj.jdbc.ha.ca.plugins;
+
+import com.mysql.cj.conf.PropertySet;
+import com.mysql.cj.jdbc.ha.ca.ClusterAwareConnectionProxy;
+import com.mysql.cj.log.Log;
+import com.mysql.cj.log.NullLogger;
+import org.junit.Assert;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import software.aws.rds.jdbc.mysql.Driver;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.spy;
+
+@TestMethodOrder(MethodOrderer.Alphanumeric.class)
+public class ConnectionPluginManagerTest {
+    @Mock IConnectionPlugin plugin;
+    @Mock ClusterAwareConnectionProxy proxy;
+    @Mock PropertySet propertySet;
+
+    private AutoCloseable openMocks;
+    private Log logger = new NullLogger(this.getClass().getName());
+
+    private static final int NUM_OF_MANAGERS = 5;
+
+    @BeforeEach
+    void setUp() {
+        openMocks = MockitoAnnotations.openMocks(this);
+        Mockito.doNothing()
+            .when(plugin).releaseResources();
+    }
+
+    @Test
+    void test_1_ReleasingAllWithEmptyList() {
+        ConnectionPluginManager.releaseAllResources();
+        Assert.assertEquals(0, ConnectionPluginManager.instances.size());
```

Comment:
1. This is not really the correct assertEquals, IntelliJ tells you that too
![image](https://user-images.githubusercontent.com/64801825/144680624-4fa982b2-1ee1-4cf2-aa5b-306f87411c5f.png)
2. Can we use static import
3. Still not sure if we need this test at all. `ConnectionPluginManager.instances.size()` is 0 to begin with in this test anyway. If I look at the implementation code of `releaseAllResources` it just has a for-each loop, if instances is empty it won't even enter the loop.  Not sure if this test adds any value, other than making sure Java for-each loop works.

Reply:
Test removed.